hydra/redisgraph_demo1.py

```diff
@@ -0,0 +1,258 @@
+
+
+"""
+
+
+import redis
+from redisgraph import Graph, Node, Edge
+import urllib
+import json
+import re
+from re import compile as regex
+from httplib2 import Http
+
+redis_con = redis.Redis(host='localhost', port=6379)
+redis_graph = Graph("apidoc", redis_con)
+"""
+getting the apidoc
+"""
+
+
+class _MemCache(dict):
+
+    def __nonzero__(self):
+        # even if empty, a _MemCache is True
+        return True
+
+    def set(self, key, value):
+        self[key] = value
+
+    def delete(self, key):
+        if key in self:
+            del self[key]
+
+
+DEFAULT_HTTP_CLIENT = Http(_MemCache())
+http = DEFAULT_HTTP_CLIENT
+APIDOC_RE = regex(
+    r'^<([^>]*)>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"$')
+
+
+def final_file(url):
+    response = urllib.urlopen(url)
+    new_file = json.loads(response.read())
+    return check_url(new_file, url)
+
+
+def check_url(new_file, url):
+    response, content = http.request(url, "GET")
+    link = response.get('link')
+    match = APIDOC_RE.match(link)
+    api_doc = match.groups()[0]
+    if api_doc != url:
+        print url, "not match", api_doc
+        new_url = api_doc
+        return final_file(new_url)
+    else:
+        return new_file
+
+
+url = "http://www.markus-lanthaler.com/hydra/event-api/"
+match_obj = re.match(r'(.*)://(.*)/(.*)/(.*)/(.*)/?', url, re.M | re.I)
+base_url = "{0}://{1}".format(match_obj.group(1), match_obj.group(2))
+entrypoint = "/" + match_obj.group(3) + "/" + match_obj.group(4) + "/"
+
+apidoc = final_file(url)
+
+"""
+extracting the endpoints and some of its properties or its api structure
+"""
+
+
+def get_endpoint_id(string):
+    return (entrypoint + string.replace("vocab:EntryPoint/", ""))
+
+
+def get_member_alias(string):
+    match_obj = re.match(r'/(.*)/(.*)/(.*)/(.*)?', string, re.M | re.I)
+#    base_url = "/{0}/{1}/".format(match_obj.group(1), match_obj.group(2))
+    entrypoint_member = "member" + match_obj.group(4)
+    return entrypoint_member
+
+
+def remove_vocab(string):
+    return string.replace("vocab:", "")
+
+
+""" Finding the entrypoint class for extracting the endpoints"""
+
+for obj in apidoc["supportedClass"]:
+    if obj["@id"] == "vocab:EntryPoint":
+        entrypoint_obj = obj
+        break
+# print entrypoint_obj["@id"]
+endpoints = {}
+endpoint_list = []
+for obj in entrypoint_obj["supportedProperty"]:
+    endpoint = {}
+    new_obj = obj["property"]
+    endpoint["title"] = obj["hydra:title"]
+    endpoint["id"] = new_obj["@id"]
+    endpoints[endpoint["title"]] = str(get_endpoint_id(endpoint["id"]))
+    endpoint_list.append(endpoint["id"])
+endpoints["@id"] = str(url)
+endpoints["@type"] = "EntryPoint"
+url_node = Node(label="id", alias="url", properties=endpoints)
+redis_graph.add_node(url_node)
+
+""" Creating endpoints as the nodes and set all the properties or
+    data with the help of endpoint """
+
+for obj in entrypoint_obj["supportedProperty"]:
```

Comment:
It is not very clear what exactly is the structure in which you are storing the API information and also not clear as to what information you are storing. As much as I can make out, you are now storing events as simple nodes and connecting all objects with a class node and the class to the collection node. Am I correct? You need to define the graph structure that you are using, maybe use a simple diagram or doc to explain what exactly is happening in the code.

Reply:
yeah, exactly I have sketched it [there](https://docs.google.com/document/d/1iQ-0UwRyrePVrmW9z7AuUyK6TiilMIimlhn9msRH3eI/edit?usp=sharing)